filter_aws: add more IMDS categories

commiterate opened the pull request:
Add more IMDS categories.

Relates to https://github.com/fluent/fluent-bit/issues/9640.

__Requires https://github.com/fluent/fluent-bit/pull/9727 to be merged.__

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Documentation**
  * Updated AWS metadata filter documentation with expanded configuration options, including new parameters for placement groups, domain settings, IPv6 support, and host identifiers, along with improved configuration examples.

<!-- review_stack_entry_start -->

[![Review Change Stack](https://storage.googleapis.com/coderabbit_public_assets/review-stack-in-coderabbit-ui.svg)](https://app.coderabbit.ai/change-stack/fluent/fluent-bit-docs/pull/1530?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

<!-- review_stack_entry_end -->

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

commiterate committed:
filter_aws: add more IMDS categories

Signed-off-by: commiterate <111539270+commiterate@users.noreply.github.com>

lockewritesdocs commented on pipeline/filters/aws-metadata.md:
```diff
@@ -9,11 +9,20 @@ The plugin supports the following configuration parameters:
+| domain | The domain for AWS resources in the region; for example, "amazonaws.com". | false |
+| region | The [region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions); for example, "us-east-1". | false |
+| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones); for example, "us-east-1a". | true |
+| az\_id | The [availability zone ID](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#az-ids); for example, "use1-az1". | false |
```
```suggestion
| az\_id | The [availability zone ID](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#az-ids), such as `"use1-az1"`. | false |
```

lockewritesdocs commented on pipeline/filters/aws-metadata.md:
```diff
@@ -9,11 +9,20 @@ The plugin supports the following configuration parameters:
+| partition | The [partition](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html); for example, "aws". | false |
+| domain | The domain for AWS resources in the region; for example, "amazonaws.com". | false |
+| region | The [region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions); for example, "us-east-1". | false |
+| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones); for example, "us-east-1a". | true |
```
```suggestion
| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-availability-zones), such as `"us-east-1a"`. | true |
```

lockewritesdocs commented on pipeline/filters/aws-metadata.md:
```diff
@@ -9,11 +9,20 @@ The plugin supports the following configuration parameters:
-| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html); for example, "us-east-1a". | true |
+| partition | The [partition](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html); for example, "aws". | false |
+| domain | The domain for AWS resources in the region; for example, "amazonaws.com". | false |
+| region | The [region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions); for example, "us-east-1". | false |
```
```suggestion
| region | The [region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions), such as `"us-east-1"`. | false |
```

lockewritesdocs commented on pipeline/filters/aws-metadata.md:
```diff
@@ -9,11 +9,20 @@ The plugin supports the following configuration parameters:
 | imds\_version | Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'. | v2 |
-| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html); for example, "us-east-1a". | true |
+| partition | The [partition](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html); for example, "aws". | false |
+| domain | The domain for AWS resources in the region; for example, "amazonaws.com". | false |
```
```suggestion
| domain | The domain for AWS resources in the region, such as `"amazonaws.com"`. | false |
```

lockewritesdocs commented on pipeline/filters/aws-metadata.md:
```diff
@@ -9,11 +9,20 @@ The plugin supports the following configuration parameters:
 | :--- | :--- | :--- |
 | imds\_version | Specify which version of the instance metadata service to use. Valid values are 'v1' or 'v2'. | v2 |
-| az | The [availability zone](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html); for example, "us-east-1a". | true |
+| partition | The [partition](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html); for example, "aws". | false |
```
```suggestion
| partition | The [partition](https://docs.aws.amazon.com/whitepapers/latest/aws-fault-isolation-boundaries/partitions.html), such as `"aws"`. | false |
```

lockewritesdocs reviewed (approved):
Thanks for your contribution @commiterate! I suggested some editorial changes from a technical writing perspective.

eschabell committed:
Merge branch 'master' into add_more_imds_categories

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/filters/aws-metadata.md:
```diff
@@ -82,26 +91,37 @@ pipeline:
+  Ec2_Instance_Type true
+  Ipv6              true
   Private_Ip        true
+  Public_Ip         true  
```
Fixing.

eschabell committed:
docs: filters: aws-metadata: fix duplicate entries and trailing whitespace

  - Remove duplicate Ec2_Instance_Id and Ec2_Instance_Type lines in classic config example
  - Remove trailing whitespace after Public_Ip value

  Applies to #1530.

Signed-off-by: Eric D. Schabell <eric@schabell.org>

eschabell commented on pipeline/filters/aws-metadata.md:
```diff
@@ -82,26 +91,37 @@ pipeline:
+  Partition_Number  true
+  Host_Id           true
+  Ec2_Instance_Id   true
+  Ec2_Instance_Type true
```
fixing.

eschabell reviewed (approved):
@commiterate just waiting on code PR merging.
